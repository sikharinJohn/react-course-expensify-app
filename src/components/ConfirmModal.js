import React from 'react';
import Modal from 'react-modal';
import { FormattedMessage } from 'react-intl';

const ConfirmModal = (props) => (
    <Modal
        isOpen={!!props.openedModal}
        onRequestClose={props.handleCloseModal}
        contentLabel={props.title}
        closeTimeoutMS={200}
        className="modal"
    >
        <FormattedMessage id="exepense-remove.title" defaultMessage="You want to remove this expense?">
            {(txt) => (
                <h3 className="modal__title">{txt}</h3>
            )}
        </FormattedMessage>
        {props.openedModal && <p className="modal__body">{props.openedModal}</p>}
        <FormattedMessage id="modal.yes" defaultMessage="Yes">
            {(txt) => (
                <button className="button button--third" onClick={props.handleConfirmModal}>{txt}</button>
            )}
        </FormattedMessage>

        <FormattedMessage id="modal.no" defaultMessage="No">
            {(txt) => (
                <button className="button button--third" onClick={props.handleCloseModal}>{txt}</button>
            )}
        </FormattedMessage>



    </Modal>
);

Modal.setAppElement('body');

export default ConfirmModal;
