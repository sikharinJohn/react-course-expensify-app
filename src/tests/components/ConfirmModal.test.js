import React from 'react';
import { shallow } from 'enzyme';
import ConfirmModal from '../../components/ConfirmModal';

test('should render ConfirmModal correctly', () => {
    const wrapper = shallow(<ConfirmModal
        title="You want to remove this expense?"
        openedModal={true}
        handleCloseModal={ () => { } }
        handleConfirmModal={ () =>{ } }
    />);
    expect(wrapper).toMatchSnapshot();
});

test('should close ConfirmModal correctly', () => {
    const wrapper = shallow(<ConfirmModal
        title="You want to remove this expense?"
        openedModal={false}
        handleCloseModal={ () => { } }
        handleConfirmModal={ () =>{ } }
    />);
    expect(wrapper).toMatchSnapshot();
});

test('should ConfirmModal on Yes button click', () => {
    const confirmModal = jest.fn();
    const wrapper = shallow(<ConfirmModal
        title="You want to remove this expense?"
        openedModal={true}
        handleCloseModal={() => { } }
        handleConfirmModal={confirmModal}
    />);
    wrapper.find('button').at(0).simulate('click');
    expect(confirmModal).toHaveBeenCalled();
});

test('should close ConfirmModal on No button click', () => {
    const closeModal = jest.fn();
    const wrapper = shallow(<ConfirmModal
        title="You want to remove this expense?"
        openedModal={true}
        handleCloseModal={closeModal}
        handleConfirmModal={ () =>{ } }
    />);
    wrapper.find('button').at(1).simulate('click');
    expect(closeModal).toHaveBeenCalled();
});