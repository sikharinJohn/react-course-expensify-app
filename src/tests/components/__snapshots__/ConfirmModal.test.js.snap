// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should close ConfirmModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="You want to remove this expense?"
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <FormattedMessage
    defaultMessage="You want to remove this expense?"
    id="exepense-remove.title"
    values={Object {}}
  />
  <FormattedMessage
    defaultMessage="Yes"
    id="modal.yes"
    values={Object {}}
  />
  <FormattedMessage
    defaultMessage="No"
    id="modal.no"
    values={Object {}}
  />
</Modal>
`;

exports[`should render ConfirmModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="You want to remove this expense?"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <FormattedMessage
    defaultMessage="You want to remove this expense?"
    id="exepense-remove.title"
    values={Object {}}
  />
  <p
    className="modal__body"
  />
  <FormattedMessage
    defaultMessage="Yes"
    id="modal.yes"
    values={Object {}}
  />
  <FormattedMessage
    defaultMessage="No"
    id="modal.no"
    values={Object {}}
  />
</Modal>
`;
